### PR TITLE
[Docs][PyOV] return back snippet for stateful model

### PR DESCRIPTION
**Pull request by akuporos**
### Details:
 - Snippet was removed here https://github.com/openvinotoolkit/openvino/issues/24510
 - Returned it back and aligned it with C++ one (generally took it from this https://github.com/openvinotoolkit/openvino/pull/15978)
 - Added it to documentation
 
### Tickets:
 - *ticket-id*

### CODE REVIEW

File: docs/articles_en/assets/snippets/ov_stateful_model_intro.py [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+            pass
+
+    #! [ov:low_latency_2_use_parameters]
+    manager.register_pass(LowLatency2(False))
+    #! [ov:low_latency_2_use_parameters]
+
+
+def replace_non_reshapable_const():
+    #! [ov:replace_const]
+    # OpenVINO example. How to replace a Constant with hardcoded values of shapes in the network with another one with the new values.
```
Comment (itikhono) [inline docs/articles_en/assets/snippets/ov_stateful_model_intro.py:84]:
```suggestion
    # OpenVINO example. How to replace a Constant with hardcoded values of shapes in the model with another one with the new values.
```

File: docs/articles_en/assets/snippets/ov_stateful_model_intro.py [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+    # Assume we know which Constant (const_with_hardcoded_shape) prevents the reshape from being applied.
+    # Then we can find this Constant by name on the network and replace it with a new one with the correct shape.
```
Comment (itikhono) [inline docs/articles_en/assets/snippets/ov_stateful_model_intro.py:86]:
```suggestion
    # Then we can find this Constant by name in the model and replace it with a new one with the correct shape.
```

File: docs/articles_en/assets/snippets/ov_stateful_model_intro.py [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+    core = ov.Core()
+    model = core.read_model("path_to_model");
+    # Creating the new Constant with a correct shape.
+    # For the example shown in the picture above, the new values of the Constant should be 1, 1, 10 instead of 1, 49, 10
+    new_const = ops.constant( """value_with_correct_shape, type""")
+    for node in model.get_ops():
+        # Trying to find the problematic Constant by name.
+        if node.get_friendly_name() != "name_of_non_reshapable_const":
+            continue
+        # Replacing the problematic Constant with a new one. Do this for all the problematic Constants in the network, then
```
Comment (itikhono) [inline docs/articles_en/assets/snippets/ov_stateful_model_intro.py:96]:
```suggestion
        # Replacing the problematic Constant with a new one. Do this for all the problematic Constants in the model, then
```

File: docs/articles_en/assets/snippets/ov_stateful_model_intro.py [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+    log.info("Loading OpenVINO")
+    core = ov.Core()
+
+    # 2. Read a model
+    log.info("Loading network files")
+    model = core.read_model("path_to_ir_xml_from_the_previous_section");
+    model.get_parameters()[0].set_layout("NC");
+    ov.set_batch(model, 1);
+
+    # 3. Load network to CPU
```
Comment (itikhono) [inline docs/articles_en/assets/snippets/ov_stateful_model_intro.py:141]:
```suggestion
    # 3. Load the model to CPU
```

File: docs/articles_en/assets/snippets/ov_stateful_model_intro.py [hunk 5]
```diff
@@ -0,0 +1,208 @@
...
+
+import openvino  as ov
+from openvino.runtime import opset13 as ops
+from openvino.runtime.op.util import VariableInfo, Variable
+from openvino.runtime.passes import LowLatency2, MakeStateful, Manager
+from openvino.runtime.utils import replace_node
+
+
+def state_network_example():
+    #! [ov:state_network]
```
Comment (itikhono) [inline docs/articles_en/assets/snippets/ov_stateful_model_intro.py:15]: it's better to rename "[ov:state_network]" to "[ov:stateful_model]"

File: docs/articles_en/assets/snippets/ov_stateful_model_intro.py [hunk 6]
```diff
@@ -0,0 +1,208 @@
...
+    input = ops.parameter([1, 1], dtype=np.float32, name="data")
+    init_const = ops.constant([[0]], dtype=np.float32)
+
+    # The ReadValue/Assign operations must be used in pairs in the network.
```
Comment (itikhono) [inline docs/articles_en/assets/snippets/ov_stateful_model_intro.py:19]:
```suggestion
    # The ReadValue/Assign operations must be used in pairs in the model.
```

Comment (itikhono) [reply docs/articles_en/assets/snippets/ov_stateful_model_intro.py:19]: "# The ReadValue/Assign operations must be used in pairs in the network." limitation is no longer relevant
Maybe we can write something like that 
"Typically ReadValue/Assign operations are presented as pairs in models. ReadValue reads information from an internal memory buffer, Assign operation writes data to this buffer. For each  pair, its own Variable object must be created. Variable defines name, shape and type of the buffer."

File: docs/articles_en/assets/snippets/ov_stateful_model_intro.py [hunk 7]
```diff
@@ -0,0 +1,208 @@
...
+    return model
+
+
+def low_latency_2_example():
+    #! [ov:low_latency_2]
+    # Precondition for Model.
+    # TensorIterator and Parameter are created in body of TensorIterator with names
+    tensor_iterator_name = "TI_name"
+    body_parameter_name = "body_parameter_name"
+    idx = "0" # this is a first variable in the network
```
Comment (itikhono) [inline docs/articles_en/assets/snippets/ov_stateful_model_intro.py:44]:
```suggestion
    idx = "0" # this is a first variable in the model
```

File: docs/articles_en/assets/snippets/ov_stateful_model_intro.py [hunk 8]
```diff
@@ -0,0 +1,208 @@
...
+
+def main():
+
+    #! [ov:state_api_usage]
+    # 1. Load inference engine
+    log.info("Loading OpenVINO")
+    core = ov.Core()
+
+    # 2. Read a model
+    log.info("Loading network files")
```
Comment (itikhono) [inline docs/articles_en/assets/snippets/ov_stateful_model_intro.py:136]:
```suggestion
    log.info("Loading model files")
```

File: docs/articles_en/assets/snippets/ov_stateful_model_intro.py (hunk 6 above)
Comment (akuporos) [reply docs/articles_en/assets/snippets/ov_stateful_model_intro.py]: updated for c++ snippet as well

File: docs/articles_en/assets/snippets/ov_stateful_model_intro.py (hunk 5 above)
Comment (akuporos) [reply docs/articles_en/assets/snippets/ov_stateful_model_intro.py]: done for c++ as well

**Review by itikhono — APPROVED** [review]

**Review by p-wysocki — APPROVED** [review]

**Review by kblaszczak-intel — APPROVED** [review]

**Pull request merged**